pkg/controller/windowsmachine/metrics/metrics.go

```diff
@@ -0,0 +1,219 @@
+		// ErrServiceMonitorNotPresent, which can be used to safely skip ServiceMonitor creation.
+		if err == metrics.ErrServiceMonitorNotPresent {
+			metricsEnabled = false
+			return errors.Wrap(err, "install prometheus-operator in your cluster to create ServiceMonitor objects")
+
+		}
+	}
+
+	oclient, err := k8sclient.NewForConfig(cfg)
+	if err != nil {
+		return errors.Wrap(err, "could not create config clientset")
+	}
+	// When a selector is present in a headless service i.e. spec.ClusterIP=None, Kubernetes manages the
+	// list of endpoints reverting all the changes made by the operator. Remove selector from Metrics Service to avoid
+	// reverting changes in the Endpoints object.
+	patchData := fmt.Sprintf(`{"spec":{"selector": null }}`)
+	service, err = oclient.CoreV1().Services(namespace).Patch(ctx, service.Name, types.MergePatchType,
+		[]byte(patchData), metav1.PatchOptions{})
+	if err != nil {
+		return errors.Wrap(err, "could not remove selector from metrics service")
+	}
+
+	return nil
+}
+
+// updateMetricsEndpoint upates the endpoint object with the new list of ip addresses from the Windows nodes.
+// It creates an endpoint object if the object is not found.
+func (pc *prometheusNodeConfig) syncMetricsEndpoint(nodeEndpointAdressess []v1.EndpointAddress) error {
+	patchData := []patchEndpoint{{
+		Op:    "replace",
+		Path:  "/subsets/0/addresses",
+		Value: nodeEndpointAdressess,
+	}}
+
+	// convert patch data to bytes
+	patchDataBytes, err := json.Marshal(patchData)
+	if err != nil {
+		return errors.Wrap(err, "unable to get patch data in bytes")
+	}
+
+	_, err = pc.k8sclientset.CoreV1().Endpoints(pc.namespace).
+		Patch(context.TODO(), windowsMetricsEndpoints, types.JSONPatchType, patchDataBytes, metav1.PatchOptions{})
+	return errors.Wrap(err, "unable to sync metrics endpoints")
+}
+
+// Configure configures patches the endpoint object to reflect the current list Windows nodes.
+func (pc *prometheusNodeConfig) Configure() error {
+	// Check if metrics are enabled in current cluster
+	if !metricsEnabled {
+		log.Info("install the prometheus-operator to enable prometheus configuration")
+		return nil
+	}
+	// get list of Windows nodes that are in Ready phase
+	nodes, err := pc.k8sclientset.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{LabelSelector: nodeconfig.WindowsOSLabel,
+		FieldSelector: "spec.unschedulable=false"})
+	if err != nil {
+		return errors.Wrap(err, "could not get Windows nodes")
+	}
+
+	// get Metrics Endpoints object
+	endpoints, err := pc.k8sclientset.CoreV1().Endpoints(pc.namespace).Get(context.TODO(),
+		windowsMetricsEndpoints, metav1.GetOptions{})
+	if err != nil {
+		return errors.Wrapf(err, "could not get metrics endpoints %v", windowsMetricsEndpoints)
+	}
+
+	if !pc.isEndpointsValid(nodes, endpoints) {
+		// check if we can get list of endpoint addresses
+		windowsIPList := pc.getNodeEndpointAddresses(nodes)
+		log.Info("get ip list", "list", windowsIPList)
```

Comment:
Do we really want to log this for the release version?

Reply:
I will remove this